src/core/providers/ogr/qgsogrprovidermetadata.cpp

```diff
@@ -316,6 +316,24 @@ QString QgsOgrProviderMetadata::encodeUri( const QVariantMap &parts ) const
   return uri;
 }
 
+QString QgsOgrProviderMetadata::absoluteToRelativeUri( const QString &uri, const QgsReadWriteContext &context ) const
+{
+  QString src = uri;
+  QStringList theURIParts = src.split( '|' );
```

Comment:
using decodeUri() and encodeUri() would avoid this method to know that | is the delimiter.
Same for relativeToAbsoluteUri()

Reply:
Indeed... there are a couple more improvements I can immediately see (e.g. handling of raster/vector virtual layer providers), but I wanted to keep this PR with least amount of changes, only move the code around, and leave logic improvements for later